opennms-config/src/main/java/org/opennms/netmgt/config/VacuumdConfigFactory.java

```diff
@@ -107,7 +103,7 @@ public VacuumdConfigFactory(VacuumdConfiguration config) {
      *                Thrown if the specified config file cannot be read
      * @throws java.io.IOException if any.
      */
-    public static synchronized void init() throws IOException {
+    public static synchronized void init() throws ConfigurationNotAvailableException {
```

Comment:
I tried to be minimal invasive here. Although the Singleton concept is a bit outdated these days, we have quite a few of them. So it would be nice if our solution would work with not many code changes...

Reply:
See bellow re: DI.